### PR TITLE
Feature/calendar styling

### PR DESCRIPTION
**Pull request by jmerz826**
## Description

This PR involves the styling of the calendar component. The design utilized the Figma markup as a great guideline, and made slight on-the-fly modifications.  Currently, the calendar is only being implemented in the instructor dashboard, but is organized in a way that it can easily be integrated to other pages in the future. This PR also addresses some rendering bugs that were previously apparent in the calendar component.

Pair-programmed with @j0lib0is via driver-navigator method!

## Loom Video

https://www.loom.com/share/f2acc70fdeea43e4a38f07512d964e52

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [ ] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

File: src/components/common/calendar/Calendar.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+import CalendarModal from './CalendarModal';
+import axiosWithAuth from '../../../utils/axiosWithAuth';
+
+const initialValues = [
+  {
+    date: '15/03/2022',
+    type: 'warning',
+    content: 'This is warning event.',
+    details: 'Test information 1',
+    time: '12:00 PM',
```
Comment (j0lib0is) [inline src/components/common/calendar/Calendar.js:14]: Good - time now rendering on badges.

File: src/components/common/calendar/Calendar.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  },
+];
+
+function CalendarApp() {
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [isScheduleModalVisible, setIsScheduleModalVisible] = useState(false);
+  const [event, setEvent] = useState(null);
+  const [eventsArr, setEventsArr] = useState(initialValues);
+
+  // ****** will update with endpoint once db endpoint is built ******
```
Comment (j0lib0is) [inline src/components/common/calendar/Calendar.js:66]: This commented code is the correct code but endpoint isn't built yet. To be uncommented when the endpoint is finished.

**Review by j0lib0is — APPROVED** [review]

**Review by MaryAngelique — APPROVED** [review]

File: src/styles/calendar.less [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+
+.ant-picker-calendar-date:hover {
+  background-color: #096a70 !important;
+  div,
+  ul span {
+    color: white !important;
+  }
+}
+
+tr:nth-of-type(6) {
```
Comment (moua0061) [inline src/styles/calendar.less:122]: This is something new I've learned with the nth-of-type. Good job of using it.

File: src/styles/calendar.less [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+.success-class-event {
+  background-color: rgba(0, 202, 183, 0.25) !important;
+}
+
+.warning-class-event {
+  background-color: #faad14 !important;
+}
+
+.error-class-event {
```
Comment (moua0061) [inline src/styles/calendar.less:76]: Nice job on adding different colors for error event, warning, and success class event to differentiate one another.

File: src/styles/calendar.less [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+    border-bottom: none !important;
+  }
+}
+
+div.ant-picker-calendar-date-value {
+  color: #f27712 !important;
+}
+
+// ---------- MODAL ----------
+.ant-picker-date-panel {
```
Comment (moua0061) [inline src/styles/calendar.less:133]: I like the idea of the modal being used when adding events to the calendar. It separates this from looking at the calendar. Good job overall on the design of the calendar.

**Review by moua0061 — APPROVED** [review]

**Review by Raj-04 — APPROVED** [review]

**Review by paulstgermain — APPROVED** [review]
WOW.

You all went above and beyond with this calendar component integration - the styling looks _incredible_ too! Merging this up now!

**Pull request merged**